tests/bayesian/bayesian_convergence_activity_minimal.py

```diff
@@ -0,0 +1,117 @@
+import random
+import matplotlib
+matplotlib.use("Agg")
+from htmresearch.frameworks.layers.l2_l4_inference import L4L2Experiment
+from htmresearch.frameworks.layers.object_machine_factory import (createObjectMachine)
+
+"""
+  Minimal version of the convergence_activity experiment from the htmpapers/frontiers repository.
+  Reproducing the experiment from the "How the neocortex learns the structure of the world" (Columns-paper).
+  Will be used to test and compare the results with the bayesian implementation of apical-TM and column pooler.
+"""
+
+def runExperiment():
+  """
+  We will run two experiments side by side, with either single column
+  or 3 columns
+  """
+  numColumns = 3
+  numFeatures = 3
+  numPoints = 10
+  numLocations = 10
+  numObjects = 10
+  numRptsPerSensation = 2
+
+  objectMachine = createObjectMachine(
+    machineType="simple",
+    numInputBits=20,
+    sensorInputSize=1024,
+    externalInputSize=1024,
+    numCorticalColumns=3,
+    seed=40,
+  )
+  objectMachine.createRandomObjects(numObjects, numPoints=numPoints,
+                                    numLocations=numLocations,
+                                    numFeatures=numFeatures)
+
+  objects = objectMachine.provideObjectsToLearn()
+
+  # single-out the inputs to the column #1
+  objectsSingleColumn = {}
+  for i in range(numObjects):
+    featureLocations = []
+    for j in range(numLocations):
+      featureLocations.append({0: objects[i][j][0]})
+    objectsSingleColumn[i] = featureLocations
+
+  # we will run two experiments side by side, with either single column
+  # or 3 columns
+  exp3 = L4L2Experiment(
```

Comment:
The L4L2 experiment is not adapted to the Bayesian approach, is it? Or does it use some other workaround to apply the Bayesian learning algorithm. If so, can me show me where it is? Otherwise we need to change that as well

Reply:
Yes it currently executes their experiment. In the next step we can adapt it to also run ours and compare results. I would not do it in this simple step,as then we would need to make the changes to the interface (e.g. compute method etc) to make sure that it passes with our new implementation. Lets do that in a next commit together with making sure it passes the interface.